Antelope-Memento/antelope_memento_api
# Fix/socket inconsistency

# Short description
_**What does this PR?**_

Fixes the inconsistency of websockets that may send transactions after 3 minutes instead of immediately and sometimes even not send them at all.

# Changes
- Emitting eventLog event updates the last checked block
- EventLog event is now emitted in one cycle with updating the state
- EventLog is now emitted using timeouts instead of intervals to prevent unexpected behaviour in the future that may have been caused if the state update took more than the update interval time (currently 500ms)

# How can it be tested
1. Start the server
2. Connect to it via websocket as in the example, providing the proper arguments. `irreversible` should be set to false or not set at all since `false` is the default.
3. Start making the transactions to the accounts you provided as arguments in websocket client. They should be sent to you almost immediately.

# Tested cases
I tested the cases of only realtime observing(`irreversible = false` and no `start_block`) and catching up to realtime and only then observing in realtime(`irreversible = false` and `start_block` set to block that is less than last irreversible block). In both cases websockets seemed to work fine, no transactions were lost.
